relayer/src/relay/messages_subscription.rs

```diff
@@ -172,28 +177,24 @@ async fn find_message_block<S: SenderConfig>(
     Ok(start_block)
 }
 
+/// Finds the first commitment where stored nonce >= 'nonce'.
+/// - sender - substrate client
+/// - network_id - eth network id
+/// - from - where to start search
+/// - nonce - nonce to compare
 async fn find_commitment_with_nonce<S: SenderConfig>(
     sender: &SubUnsignedClient<S>,
     network_id: GenericNetworkId,
-    count: u32,
+    from_block: BlockNumber<S>,
     nonce: u64,
 ) -> AnyResult<Option<(BlockNumber<S>, H256)>> {
-    let start_block = find_message_block(sender, network_id, nonce).await?;
-    let start_block = if let Some(start_block) = start_block {
-        start_block + 1u32.into()
-    } else {
-        return Ok(None);
-    };
-    for i in 0..count {
```

Comment:
This function is also used by substrate bridge which is currently use message nonces and can miss commitment